apps/root/src/services/earnService.ts

```diff
@@ -527,11 +528,30 @@ export class EarnService extends EventsManager<EarnServiceData> {
           amount: amount,
         };
 
+    const earnCompanionAddress = this.contractService.getEarnCompanionAddress(strategy.farm.chainId);
+
+    const permissions = [
+      {
+        operator: earnCompanionAddress,
+        permissions: [EarnPermission.INCREASE],
+      },
+    ];
+
+    const wrappedProtocol = getWrappedProtocolToken(strategy.farm.chainId);
+
+    if (strategy.farm.asset.address === wrappedProtocol.address) {
+      permissions.push({
+        operator: earnCompanionAddress,
+        permissions: [EarnPermission.WITHDRAW],
```

Comment:
Why add this on position creation?

Reply:
Its going to need it anyway if he wants to withdraw with it, we are saving a step by doing it here